### ISSUE TITLE
assoc with multiple keys

### ISSUE THREAD

**Issue by jrbrodie77** [pytoolz/toolz#402]
I'm wondering if there's a reason that assoc only takes one k,v pair?  Considering that the function does copy on write, it seems like it would be efficient to do:

assoc(mydict, k1, v1, k2, v2..., factory=dict)

I suppose that this would be a breaking change for people not using the keyword in the factory arg.

**Comment by eriknw** [comment]
This seems reasonable to me and is similar to `dict.update`.

**Comment by jrbrodie77** [comment]
I don't think that the first case of (k1, k2...), (v1, v2...) is very elegant.  One problem is that tuples are valid dict keys, so this conflicts with simple k0, v0 cases (how it works now).

I like the 'items' case with ((k1, v1), (k2, v2)).  This supports using dict.items() output. Also preserves order if factory is OrderedDict.

The case with assoc(mydict, <dict>) duplicates merge, so perhaps leave that one out based on a the 'one way to do something' Zen of Python edict?

One additional note: the documentation for merge doesn't mention using a factory, but I note that the implementation of assoc calls merge with a factory.  Would be a nice addition to the docs if someone already has a PR going.
